jwst/resample/gwcs_drizzle.py

```diff
@@ -55,70 +48,56 @@ def __init__(self, product, outwcs=None, wt_scl=None,
             The value a pixel is set to in the output if the input image does
             not overlap it. The default value of NAN sets NaN values.
         """
-
         # Initialize the object fields
         self._product = product
-        self.outsci = None
-        self.outwht = None
-        self.outcon = None
-        self.uniqid = 0
 
-        if wt_scl is None:
-            self.wt_scl = ""
-        else:
-            self.wt_scl = wt_scl
-        self.kernel = kernel
-        self.fillval = fillval
         self.pixfrac = pixfrac
 
-        self.sciext = "SCI"
-        self.whtext = "WHT"
-        self.conext = "CON"
-
-        out_units = "cps"
-
         self.outexptime = product.meta.exposure.measurement_time or 0.0
 
         self.outsci = product.data
+        self.outwht = product.wht
+
         if outwcs:
             self.outwcs = outwcs
         else:
             self.outwcs = product.meta.wcs
 
-        self.outwht = product.wht
-        self.outcon = product.con
-
-        if self.outcon.ndim == 2:
-            self.outcon = np.reshape(self.outcon, (1,
-                                                   self.outcon.shape[0],
-                                                   self.outcon.shape[1]))
-        elif self.outcon.ndim != 3:
-            raise ValueError("Drizzle context image has wrong dimensions: \
-                {0}".format(product))
-
         # Check field values
         if not self.outwcs:
             raise ValueError("Either an existing file or wcs must be supplied")
 
-        if out_units == "counts":
-            np.divide(self.outsci, self.outexptime, self.outsci)
-        elif out_units != "cps":
-            raise ValueError("Illegal value for out_units: %s" % out_units)
-
-    # Since the context array is dynamic, it must be re-assigned
-    # back to the product's `con` attribute.
-    @property
-    def outcon(self):
-        """Return the context array"""
-        return self._product.con
-
-    @outcon.setter
-    def outcon(self, value):
-        """Set new context array"""
-        self._product.con = value
-
-    def add_image(self, insci, inwcs, inwht=None, xmin=0, xmax=0, ymin=0, ymax=0,
-                  expin=1.0, in_units="cps", wt_scl=1.0, iscale=1.0):
+        ctx = product.con
```

Comment:
The `product.con` access here will create a context array (if one doesn't already exist). Putting it in an `if not disable_ctx` block (and similarly guarding other `product.con` accesses would prevent this). See https://github.com/spacetelescope/jwst/issues/8638#issuecomment-2422454499 for more details.